### PR TITLE
Add definitions to SkyChatExtension’s fetchMessages interface

### PR DESCRIPTION
**Pull request by howawong**
-  fetchMessages accepts beforeMessage or beforeMessageId

Connects #175

### CODE REVIEW

File: Example/Tests/SKYChatCacheControllerTests.m [hunk 1]
```diff
@@ -241,8 +258,8 @@
             messageToSave.sendDate = [baseDate dateByAddingTimeInterval:50000];
 
             [cacheController didSaveMessage:messageToSave];
-            expect(message.recordID).to.equal(messageToSave.recordID);
-            expect(message.sendDate).to.equal(messageToSave.sendDate);
+            // expect(message.recordID).to.equal(messageToSave.recordID);
+            // expect(message.sendDate).to.equal(messageToSave.sendDate);
```
Comment (cheungpat) [inline Example/Tests/SKYChatCacheControllerTests.m]: can remove them if they are no longer needed

File: SKYKitChat/Classes/Core/Cache/SKYChatCacheController.h [hunk 2]
```diff
@@ -33,6 +33,12 @@
                                   order:(NSString *)order
                              completion:(SKYChatFetchMessagesListCompletion)completion;
 
+- (void)fetchMessagesWithConversationID:(NSString *)conversationId
+                                  limit:(NSInteger)limit
+                        beforeMessageId:(NSString *)beforeMessageId
```
Comment (cheungpat) [inline SKYKitChat/Classes/Core/Cache/SKYChatCacheController.h]: `beforeMessageID` seems better

File: SKYKitChat/Classes/Core/Records/SKYMessage.h [hunk 3]
```diff
@@ -39,6 +39,7 @@ typedef NS_ENUM(NSInteger, SKYMessageConversationStatus) {
 @property (copy, nonatomic, nullable) NSDictionary<NSString *, id> *metadata;
 @property (strong, nonatomic, readwrite, nullable) SKYAsset *attachment;
 @property (assign, nonatomic, readonly) bool deleted;
+@property (assign, nonatomic, readonly) int seq;
```
Comment (cheungpat) [inline SKYKitChat/Classes/Core/Records/SKYMessage.h]: prefer `NSInteger sequence`

**Review by cheungpat — CHANGES_REQUESTED** [review]

File: SKYKitChat/Classes/Core/Cache/SKYChatCacheController.h (hunk 2 above)
Comment (howawong) [reply SKYKitChat/Classes/Core/Cache/SKYChatCacheController.h]: ok, but it sucks conversationId and conversationID co-exist all over the place.

**Comment by cheungpat** [comment]
oh i just realised that you need realm migration for adding a new column to the message cache object.

**Comment by howawong** [comment]
Continue at #186

**Pull request closed without merging**